src/Cli/dotnet/SlnFileFactory.cs

```diff
@@ -10,6 +11,35 @@ namespace Microsoft.DotNet.Tools.Common
 {
     public static class SlnFileFactory
     {
+        public static string GetSolutionFileFullPath(string slnFileOrDirectory, bool includeSolutionFilterFiles = false, bool includeSolutionXmlFiles = true)
+        {
+            if (File.Exists(slnFileOrDirectory))
```

Comment:
It occurred to me that if a user can pass slnFileOrDirectory in, this could turn into an ugly exception with a stack trace if they pass an invalid path, and the resulting exception isn't caught. Is there a catch somewhere?

Reply:
You mean making sure the passed string is a valid path?

Reply:
Yes